src/graph/components/DagNode.tsx

```diff
@@ -16,30 +17,60 @@ export const DagNode = memo(
     targetPosition,
     sourcePosition,
   }: NodeProps<DataPayload>): JSX.Element => {
-    const { selectedTableId, selectedColumnId, reletedColumnsIds, setHighlightedColumns } =
-      useContext(HighlightedColumnsContext)
+    const {
+      selectedTableId,
+      selectedColumnId,
+      reletedColumnsIds,
+      setHighlightedColumns,
+      highlightedType,
+    } = useContext(HighlightedColumnsContext)
 
     const actualBadge = badge === 'violations' ? violations : partitions
     const actualBadgeFormatted = renderNumericValue(2, true)(actualBadge)
     const _columns =
-      selectedTableId && id !== selectedTableId
-        ? columns.filter(column => reletedColumnsIds.includes(column.id))
+      highlightedType !== 'none' &&
+      highlightedType !== 'diffing' &&
+      selectedTableId &&
+      id !== selectedTableId
+        ? columns.filter((column: any) => reletedColumnsIds.includes(column.id))
         : columns
 
-    const getListItemHighlightedColor = (columnId: string): string => {
-      if (id === selectedTableId && columnId === selectedColumnId) {
+    const getListItemHighlightedColor = (column: Column): string => {
+      if (id === selectedTableId && column.id === selectedColumnId) {
         return 'rgb(0, 0, 0)'
       }
 
       if (id === selectedTableId) {
         return 'rgba(0, 0, 0, 0.4)'
       }
 
-      return style?.color ? style.color : 'inherit'
+      if (
+        style?.color === ADDED_NODE_HIGHLIGHT_COLOR ||
+        style?.color === DELETED_NODE_HIGHLIHT_COLOR
+      ) {
+        return style.color
+      }
+
+      if (column?.style?.color) {
+        return column.style.color
+      }
+
+      return 'inherit'
```

Comment:
You might like more these `css` parts to be inside of a style component

Reply:
It is either way difficult to get what's the meaning behind 'inherit' and 'rgba(0, 0, 0, 0.4)' 
They have to be constantsat least